### PR TITLE
Remove base payment component

### PR DESCRIPTION
**Pull request by jreij**
## Description
Now that we moved the logic outside of the `BasePaymentComponent` abstract class we can safely remove it.

## Checklist 
- [x] Changes are tested manually

COAND-684

### CODE REVIEW

File: components-core/src/main/java/com/adyen/checkout/components/base/BasePaymentComponent.kt [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-
-import android.content.Context
-import androidx.lifecycle.SavedStateHandle
-import com.adyen.checkout.components.PaymentComponentState
-import com.adyen.checkout.components.base.lifecycle.PaymentComponentViewModel
-import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails
-import com.adyen.checkout.core.log.LogUtil
-
-@Suppress("TooManyFunctions")
-abstract class BasePaymentComponent<
```
[the remark this answers is not in the document]
Comment (etman55) [reply components-core/src/main/java/com/adyen/checkout/components/base/BasePaymentComponent.kt:18]: Finally 🥳

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/stored/PreselectedStoredPaymentMethodFragment.kt [hunk 2]
```diff
@@ -58,7 +57,7 @@ internal class PreselectedStoredPaymentMethodFragment : DropInBottomSheetDialogF
     private val binding: FragmentStoredPaymentMethodBinding get() = requireNotNull(_binding)
     private lateinit var storedPaymentMethod: StoredPaymentMethod
     private lateinit var imageLoader: ImageLoader
-    private lateinit var component: PaymentComponent<PaymentComponentState<in PaymentMethodDetails>, Configuration>
+    private lateinit var component: PaymentComponent<out PaymentComponentState<out PaymentMethodDetails>>
```
[the remark this answers is not in the document]
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/stored/PreselectedStoredPaymentMethodFragment.kt:60]: What's the alternative? `*`?

File: action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt [hunk 3]
```diff
@@ -29,7 +29,7 @@ import com.adyen.checkout.wechatpay.WeChatPayActionComponent
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 fun getActionProviderFor(
     action: Action
-): ActionComponentProvider<out ActionComponent<out Configuration>, out Configuration, out ActionDelegate>? {
+): ActionComponentProvider<out ActionComponent, out Configuration, out ActionDelegate>? {
```
Comment (OscarSpruit) [inline action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt:32]: Does `*` also work here?

File: components-core/src/main/java/com/adyen/checkout/components/PaymentComponent.kt [hunk 4]
```diff
@@ -19,10 +18,7 @@ import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails
  *
  * Should be used attached to a corresponding ComponentView to get data from.
  */
-interface PaymentComponent<
-    ComponentStateT : PaymentComponentState<out PaymentMethodDetails>,
-    ConfigurationT : Configuration> :
-    Component<ComponentStateT, ConfigurationT> {
+interface PaymentComponent<ComponentStateT : PaymentComponentState<out PaymentMethodDetails>> : Component {
```
Comment (OscarSpruit) [inline components-core/src/main/java/com/adyen/checkout/components/PaymentComponent.kt:21]: And here? `*`

File: action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt (hunk 3 above)
Comment (jreij) [reply action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt:32]: Yes indeed 👍

File: components-core/src/main/java/com/adyen/checkout/components/PaymentComponent.kt (hunk 4 above)
Comment (jreij) [reply components-core/src/main/java/com/adyen/checkout/components/PaymentComponent.kt:21]: Yep

File: action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt (hunk 3 above)
Comment (jreij) [reply action/src/main/java/com/adyen/checkout/action/ActionComponentExtensions.kt:32]: Yes

**Review by OscarSpruit — APPROVED** [review]
Great cleanup!

**Pull request merged**